src/backends/gpu/mechanism.cpp

```diff
@@ -51,6 +51,8 @@ void mechanism::instantiate(fvm_size_type id, backend::shared_state& shared, con
 
     mechanism_ppack_base* pp = ppack_ptr(); // From derived class instance.
 
+    pp->width_ = width_;
+
```

Comment:
The 2/3 letter variable name scheme appears to breaking down in this file:
Trying to understand what is going on here I found:
ion_
n_ion
oion
Without really being clear if these are class members or local function members.
I would suggest helping the reader of the code by using consistent informative names -or- source code comments.

Reply:
I had a look for the problem, and found the following
  * there is no variable named `ion_` (or `ion`)
  * there is no variable named `n_ion`, There is one named `n_ion_` (a class member because it has the trailing `_`)
  * `oion` is an `util::optional` wrapper, and is used in the 5 lines below where it is returned from a function.

We have standardised practice of using a trailing underscore, e.g. `n_ion_`, to indicate class members, which is being followed in this class.

I renamed `n_ion_` and `n_field` to `num_ions_` and `num_fields` respectively, to make their meaning a bit clearer.

Reply:
I can't argue against greater clarity, but I will assert `n_foo` is a perfectly reasonable way of naming a variable that holds the number of foos.